viewer/packages/cad-model/src/picking/PickingHandler.ts

```diff
@@ -123,9 +120,10 @@ export class PickingHandler {
         // Make current CadNode visible & hide others
         visibleCadNodes.forEach(p => (p.visible = false));
         cadNodeData.cadNode.visible = true;
-
         const treeIndex = await this.intersectCadNodeTreeIndex(cadNodeData.cadNode, input, async);
         if (treeIndex) {
+          // Assuming we have depth anywhere we hit a treeIndex
+          const depthResult = await this.intersectCadNodeDepth(depthInput, async);
           const result: IntersectCadNodesResult = {
             distance: depthResult.distance,
             point: depthResult.point,
```

Comment:
In the old code the reuse of the depthResult here will cause the `return results.sort((l, r) => l.distance - r.distance);` to just return a node, independent of real distance. If two models overlap this may not be the closest treeIndex.